dsgrid/registry/dimension_mapping_registry_manager.py

```diff
@@ -144,17 +140,92 @@ def _check_records_against_dimension_records(self, config):
             if diff:
                 dim_id = from_dimension.model.dimension_id
                 raise DSGInvalidDimensionMapping(
-                    f"Dimension mapping has invalid 'to' records: dimension_id={dim_id} {diff}"
+                    f"Dimension mapping={mapping.filename} has invalid 'to_id' records: {diff}, "
+                    f"they are missing from dimension_id={dim_id}"
                 )
 
     def validate_records(self, config: DimensionMappingsConfig, warn_only=False):
         """Validate dimension mapping records.
 
         Check:
-        - from_id and to_id column names
-        - check for duplicate IDs and log a warning if they exist (sometimes we want them to exist if it is an aggregation)
+        - duplicate records in from_id and to_id columns per mapping archetype
+        - sum of from_fraction by from_id per mapping archetype
+
         """
-        pass
+        for mapping in config.model.mappings:
+            if "one_to_one" in mapping.archetype.value:
+                allow_dup_from_records = False
+                allow_dup_to_records = False
+            elif "one_to_many" in mapping.archetype.value:
+                allow_dup_from_records = True
+                allow_dup_to_records = False
+            elif "many_to_one" in mapping.archetype.value:
+                allow_dup_from_records = False
+                allow_dup_to_records = True
+            elif "many_to_many" in mapping.archetype.value:
+                allow_dup_from_records = True
+                allow_dup_to_records = True
+
+            actual_from_records = [x.from_id for x in mapping.records]
+            self._check_for_duplicates_in_list(
+                actual_from_records,
+                allow_dup_from_records,
+                "from_id",
+                mapping.filename,
+                mapping.mapping_type.value,
+            )
+            actual_to_records = [x.to_id for x in mapping.records]
+            self._check_for_duplicates_in_list(
+                actual_to_records,
+                allow_dup_to_records,
+                "to_id",
+                mapping.filename,
+                mapping.mapping_type.value,
+            )
+
+            if "fraction_sum_eq1" in mapping.archetype.value:
+                self._check_fraction_sum(
+                    mapping.records, mapping.filename, mapping.mapping_type.value
+                )
```

Comment:
If had SUM_GT1 and SUM_LT1 archetypes could also check for that here when applicable